pkg/helmreconciler/pruning.go

```diff
@@ -28,27 +29,20 @@ import (
 
 // Prune removes any resources not specified in manifests generated by HelmReconciler h. If all is set to true, this
 // function prunes all resources.
-func (h *HelmReconciler) Prune(all bool) error {
+func (h *HelmReconciler) Prune(excluded map[string]bool, all bool) error {
 	allErrors := []error{}
-	namespacedResources, nonNamespacedResources := h.customizer.PruningDetails().GetResourceTypes()
+	namespacedResources, _ := h.customizer.PruningDetails().GetResourceTypes()
 	targetNamespace := h.customizer.Input().GetTargetNamespace()
-	err := h.PruneResources(namespacedResources, all, targetNamespace)
-	if err != nil {
-		allErrors = append(allErrors, err)
-	}
-	err = h.PruneResources(nonNamespacedResources, all, "")
```

Comment:
we still need to prune all the `nonNamespacedResources`,  isn't it?

Reply:
No, we decided to leak the common resources so this was actually incorrect. The reason is if you need to rollback, the deleted CRDs will delete the CRs and this will make it impossible to restore the previous state.

Reply:
We shouldn't be deleting CRDs, but should be deleting the other cluster scoped resources (e.g. ClusterRole, ClusterRoleBinding, etc.).

Reply:
Are non-CRD cluster scoped resources shared between ICP resources? Despite the name, I think our ClusterRole/ClusterRoleBindings are instantiated per-namespace (the namespace is encoded in the name). Removing them might be okay, but it does grant the operator broader cluster-wide privileges vs. adding/removing out of band.

Reply:
The namespaced counter parts are Role and RoleBinding.  Also for multi-tenancy, where user namespaces opt-in to the mesh, we (maistra) still use ClusterRole resources as it simplifies management of the RoleBinding resources that get copied into the member namespaces.

That said, while they are not shared, we should be cleaning them up.  (FWIW, I don't consider CRDs to be part of a control plane install, but rather a bootstrapping operation done by the operator, as those are shared/required by all installations.)